### PR TITLE
Add `Expr::Name` checks to rules which use `is_logger_candidate` 

### PR DESCRIPTION
**Pull request by qdegraaf**
## Summary

Expands several rules to also check for `Expr::Name` values. As they would previously not consider:
```python
from logging import error

error("foo")
```
as potential violations
```python
import logging

logging.error("foo")
```
as potential violations leading to inconsistent behaviour. 

The rules impacted are:

- `BLE001`
- `TRY400`
- `TRY401`
- `PLE1205`
- `PLE1206`
- `LOG007`
- `G001`-`G004`
- `G101`
- `G201`
- `G202`

## Test Plan

Fixtures for all impacted rules expanded. 

## Issue Link

Refers: https://github.com/astral-sh/ruff/issues/7502

### CODE REVIEW

**Comment by charliermarsh** [comment]
So I think there's a little more to do here, but it's not clear from this PR because there's actually a bug in `LOG007`. Lemme push a fix, then I'll come back and explain.

**Comment by charliermarsh** [comment]
Take a look at https://github.com/astral-sh/ruff/pull/7524. The issue is that we assume all calls are done on attributes, so that we can resolve the attribute to, e.g., `.exception` or `.info` or whatever. The same pattern is used in `logging_call`, in `crates/ruff/src/rules/flake8_logging_format/rules/logging_call.rs`.

We may want to change `logging_call` (and the `LOG007` method) to match on either name or attribute, and use separate paths to determine whether it's a logger and, if so, what the log level is. _Or_ we could try changing _this_ method to return the logging attribute too.

**Comment by qdegraaf** [comment]
Interesting, thanks for the taking the time to explain and clarifying my previous misinterpretation of `is_logger_candidate` in `LOG007`. I'll have a look tomorrow if I can implement what you're suggesting. Will mark as draft until then.

File: crates/ruff_linter/src/rules/flake8_logging_format/rules/logging_call.rs [hunk 1]
```diff
@@ -153,22 +153,36 @@ impl LoggingCallType {
...
+                &call.func,
+                checker.semantic(),
+                &checker.settings.logger_objects,
+            ) {
+                return;
+            }
+            (call_type, attr.range())
+        }
+        Expr::Name(_) => {
+            let Some(call_path) = checker.semantic().resolve_call_path(call.func.as_ref()) else {
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_logging_format/rules/logging_call.rs:172]: @qdegraaf - I changed this to always use `resolve_call_path`. I think we want this to _only_ match the logging imports.

**Comment by charliermarsh** [comment]
@qdegraaf - Nice! I think we need to extend this to a few more call sites. Are you able to look into the other usages of `is_logger_candidate` and apply similar logic? E.g., in `blind_except`, in the callers of `LoggerCandidateVisitor`, and so on. If not, we can merge and fix those up later, but I believe they do also need refinement.

**Comment by qdegraaf** [comment]
Happy to add those to the PR! Will have a look tomorrow

**Comment by charliermarsh** [comment]
Thank you!

**Review by charliermarsh — APPROVED** [review]

**Pull request merged**